ui/pages/confirmations/components/confirm/info/shared/transaction-details.tsx

```diff
@@ -0,0 +1,54 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { isValidAddress } from 'ethereumjs-util';
+import React from 'react';
+import { useSelector } from 'react-redux';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowAddress,
+  ConfirmInfoRowText,
+  ConfirmInfoRowUrl,
+} from '../../../../../../components/app/confirm/info/row';
+import { useI18nContext } from '../../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../selectors';
+import { useKnownMethodDataInTransaction } from '../../hooks';
+
+export const TransactionDetails = () => {
+  const t = useI18nContext();
+
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
+
+  const { knownMethodData } =
+    useKnownMethodDataInTransaction(currentConfirmation);
+
+  return (
+    <>
+      {currentConfirmation?.origin ? (
```

Comment:
This is only a moderate size for now, but in the interest of readability and scalability, could we define some local functions to provide sub-components for each conceptual piece of the component?

Such as `OriginRow`, `ToRow`, and `MethodDataRow`?

We could even move the use of the method data hook to the `MethodDataRow` function to encapsulate the concept?